### PR TITLE
Add support new contracts

### LINKED ISSUE #7: Add support of new contracts
As part of revisiting bridge contracts and adding support of `erc20-to-erc20` bridge mode (https://github.com/poanetwork/poa-bridge-contracts/tree/refactor_v1) it is necessary to investigate what set of changes needs to done on side of `bridge-monitor` and create set of issues in GitHub to address these changes.

### PR DESCRIPTION
**Pull request by patitonar**
Created `develop` branch which is the target of this PR, updated abis files, events names and added logic for `erc20-to-erc20` mode. To detect the bridge mode I used the same logic as in Bridge-UI suggested [here](https://github.com/poanetwork/bridge-ui/pull/135#discussion_r212229299).

Closes #7

### CODE REVIEW

File: abis/BridgeValidators.abi.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "anonymous": false,
     "inputs": [
       {
-        "indexed": false,
+        "indexed": true,
```
[the remark this answers is not in the document]
Comment (fvictorio) [reply abis/BridgeValidators.abi.json]: That would be great, and something we should do in `bridge-nodejs` too (we already have the contracts as a submodule, but the abis are still hardcoded in the `abis` directory).

**Comment by patitonar** [comment]
Here are some example outputs

**Native-to-Erc20** - deployed on Sokol - Kovan

[   Home  ] HomeBridge: 0xFa143Ad92c6E5307c71EFEf2079EDb84504A1a11
[ Foreign ] ForeignBridge: 0x0842dfF263672cBE89dD210aD8E3fBE4cb57e181
[ Foreign ] POA20: 0xFcD624399f6fAb83f142ddd249Fd3f9EAA6388Dc

`/validators`
```
{
    "home": {
        "validators": {
            "0x3752F0a521b2E383693f658D6FC377953024C207": {
                "balance": "10.895701392085",
                "leftTx": 36319004640283,
                "gasPrice": 1
            }
        }
    },
    "foreign": {
        "validators": {
            "0x3752F0a521b2E383693f658D6FC377953024C207": {
                "balance": "2.654112996226",
                "leftTx": 2949,
                "gasPrice": 3
            }
        }
    },
    "lastChecked": 1536154409
}
```

Scenario: 
    - 2 deposits on Home, only 1 was bridged
    - 1 withdraw on Foreign not bridged 

`/`
```
{
    "home": {
        "balance": "0.02",
        "deposits": 2,
        "withdrawals": 0,
        "requiredSignatures": 1
    },
    "foreign": {
        "totalSupply": "0",
        "deposits": 1,
        "withdrawals": 1,
        "requiredSignatures": 1
    },
    "balanceDiff": 0.02,
    "lastChecked": 1536155375,
    "depositsDiff": 1,
    "withdrawalDiff": -1,
    "requiredSignaturesMatch": true
}
```

`/eventsStats`
```
{
    "onlyInHomeDeposits": [
        {
            "address": "0xFa143Ad92c6E5307c71EFEf2079EDb84504A1a11",
            "blockHash": "0x4edc1da670867af5b7521cffa87a455d1c3079e2f3ad4d6a163f6d2551b3f0dc",
            "blockNumber": 4389719,
            "logIndex": 0,
            "removed": false,
            "transactionHash": "0xad1a663b68718bfa2ebdcbd2aa9a1b0ec6bde2a303100c23c64dc2bcd3164eb9",
            "transactionIndex": 1,
            "transactionLogIndex": "0x0",
            "type": "mined",
            "id": "log_963187a3",
            "returnValues": {
                "0": "0xf4BEF13F9f4f2B203FAF0C3cBbaAbe1afE056955",
                "1": "10000000000000000",
                "recipient": "0xf4BEF13F9f4f2B203FAF0C3cBbaAbe1afE056955",
                "value": "10000000000000000"
            },
            "event": "UserRequestForSignature",
            "signature": "0x127650bcfb0ba017401abe4931453a405140a8fd36fece67bae2db174d3fdd63",
            "raw": {
                "data": "0x000000000000000000000000f4bef13f9f4f2b203faf0c3cbbaabe1afe056955000000000000000000000000000000000000000000000000002386f26fc10000",
                "topics": [
                    "0x127650bcfb0ba017401abe4931453a405140a8fd36fece67bae2db174d3fdd63"
                ]
            }
        }
    ],
    "onlyInForeignDeposits": [],
    "onlyInHomeWithdrawals": [],
    "onlyInForeignWithdrawals": [
        {
            "address": "0x0842dfF263672cBE89dD210aD8E3fBE4cb57e181",
            "blockHash": "0xb6a5c867828387db9a6eebaee9d87af888c62fab9a94599190d5d838b8999fee",
            "blockNumber": 8617562,
            "logIndex": 5,
            "transactionHash": "0x43bc0a90ed484667ac9ea74c619bf944b90e318f1ea00bb20055304a46adb6ac",
            "transactionIndex": 3,
            "transactionLogIndex": "0x4",
            "type": "mined",
            "id": "log_f111379a",
            "returnValues": {
                "0": "0xbb140FbA6242a1c3887A7823F7750a73101383e3",
                "1": "10000000000000000",
                "recipient": "0xbb140FbA6242a1c3887A7823F7750a73101383e3",
                "value": "10000000000000000"
            },
            "event": "UserRequestForAffirmation",
            "signature": "0x1d491a427d1f8cc0d447496f300fac39f7306122481d8e663451eb268274146b",
            "raw": {
                "data": "0x000000000000000000000000bb140fba6242a1c3887a7823f7750a73101383e3000000000000000000000000000000000000000000000000002386f26fc10000",
                "topics": [
                    "0x1d491a427d1f8cc0d447496f300fac39f7306122481d8e663451eb268274146b"
                ]
            }
        }
    ],
    "lastChecked": 1536155528
}
```



**ERC20-to-ERC20** - deployed on Sokol - Kovan

[   Home  ] HomeBridge: 0xdB6D47aeBa29E0f128810dD80C9eE72AA2301d27
[ Foreign ] ForeignBridge: 0x140c1e7C068E6CDD07293B7Cf327718F0bFBf079
[ Foreign ] ERC20 Token: 0x630d2C61234224fD9705457BE61B830A0Ea81822
[ Home ] ERC677 Bridgeble Token: 0x9b0A0dCa48358840deD0fccBD2154f354678285F

`/validators`
```
{
    "home": {
        "validators": {
            "0x3752F0a521b2E383693f658D6FC377953024C207": {
                "balance": "10.895701392085",
                "leftTx": 36319004640283,
                "gasPrice": 1
            }
        }
    },
    "foreign": {
        "validators": {
            "0x3752F0a521b2E383693f658D6FC377953024C207": {
                "balance": "2.654112996226",
                "leftTx": 2949,
                "gasPrice": 3
            }
        }
    },
    "lastChecked": 1536157625
}
```

Scenario: 
    - 1 deposits on Home bridged
    - 3 withdraw on Foreign, 1 not bridged 

`/`
```
{
    "home": {
        "totalSupply": "0.01",
        "deposits": 1,
        "withdrawals": 2,
        "requiredSignatures": 1
    },
    "foreign": {
        "balance": "0",
        "deposits": 1,
        "withdrawals": 3,
        "requiredSignatures": 1
    },
    "balanceDiff": -0.01,
    "lastChecked": 1536158465,
    "depositsDiff": 0,
    "withdrawalDiff": -1,
    "requiredSignaturesMatch": true
}
```

`/eventsStats`
```
{
    "onlyInHomeDeposits": [],
    "onlyInForeignDeposits": [],
    "onlyInHomeWithdrawals": [],
    "onlyInForeignWithdrawals": [
        {
            "address": "0x630d2C61234224fD9705457BE61B830A0Ea81822",
            "blockHash": "0xd58b5847b4c673488897461de671273019c1a07a1103ec41208c8442f5624478",
            "blockNumber": 8617983,
            "logIndex": 0,
            "transactionHash": "0x546c5eab95043564f6c7f2a8d14241f6769e151b0ff5a05dffcf66409a3a0d6f",
            "transactionIndex": 0,
            "transactionLogIndex": "0x0",
            "type": "mined",
            "id": "log_79d42824",
            "returnValues": {
                "0": "0x3752F0a521b2E383693f658D6FC377953024C207",
                "1": "0x140c1e7C068E6CDD07293B7Cf327718F0bFBf079",
                "2": "10000000000000000",
                "from": "0x3752F0a521b2E383693f658D6FC377953024C207",
                "to": "0x140c1e7C068E6CDD07293B7Cf327718F0bFBf079",
                "value": "10000000000000000"
            },
            "event": "Transfer",
            "signature": "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
            "raw": {
                "data": "0x000000000000000000000000000000000000000000000000002386f26fc10000",
                "topics": [
                    "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
                    "0x0000000000000000000000003752f0a521b2e383693f658d6fc377953024c207",
                    "0x000000000000000000000000140c1e7c068e6cdd07293b7cf327718f0bfbf079"
                ]
            }
        }
    ],
    "lastChecked": 1536158571
}
```


Right now on ERC20-to-ERC20 mode on Foreign information the balance of the contract is displayed which is not very useful, I think instead we should display the ERC20 balance of the Foreign bridge, which in the example from above would be `0.01`. Does this makes sense or do you have any other suggestion for this @akolotov ?

File: abis/BridgeValidators.abi.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (patitonar) [reply abis/BridgeValidators.abi.json]: Created https://github.com/poanetwork/bridge-monitor/issues/10

File: getBalances.js [hunk 2]
```diff
@@ -17,28 +16,55 @@ const web3Home = new Web3(homeProvider);
...
+    if (isErcToErcMode) {
+      logger.debug('calling web3Foreign.eth.getBalance');
+      const foreignBalance = await web3Foreign.eth.getBalance(FOREIGN_BRIDGE_ADDRESS)
+      const homeBridge = new web3Home.eth.Contract(HOME_ERC_ABI, HOME_BRIDGE_ADDRESS);
+      logger.debug('calling homeBridge.methods.erc677token');
+      const tokenAddress = await homeBridge.methods.erc677token().call()
+      const tokenContract = new web3Home.eth.Contract(ERC677_ABI, tokenAddress);
+      logger.debug('calling tokenContract.methods.totalSupply()');
+      const totalSupply = await tokenContract.methods.totalSupply().call()
+      const foreignBalanceBN = new BN(foreignBalance)
```
Comment (akolotov) [inline getBalances.js]: Yes, you are right (to the note at the end of https://github.com/poanetwork/bridge-monitor/pull/9#issuecomment-418765582). For `erc20-to-erc20` mode we need to monitor balance of ERC20 for the foreign bridge contract.

Comment (patitonar) [reply getBalances.js]: @akolotov Updated to display balance of ERC20 for the foreign bridge contract.
Example of output for erc20-erc20 mode:

```
{
    "home": {
        "totalSupply": "0.01",
        "deposits": 1,
        "withdrawals": 2,
        "requiredSignatures": 1
    },
    "foreign": {
        "erc20Balance": "0.02",
        "deposits": 1,
        "withdrawals": 3,
        "requiredSignatures": 1
    },
    "balanceDiff": 0.01,
    "lastChecked": 1536325837,
    "depositsDiff": 0,
    "withdrawalDiff": -1,
    "requiredSignaturesMatch": true
}
```

**Comment by akolotov** [comment]
We could merge this PR

**Pull request merged**